addons/website_sale_slides/views/website_sale_templates.xml

```diff
@@ -3,13 +3,14 @@
 
 <template id="website_sale_confirmation_slide" inherit_id="website_sale.confirmation">
     <xpath expr="//div[@id='oe_structure_website_sale_confirmation_2']" position="after">
-        <t t-call="website_sale_slides.course_purchased_confirmation_message"/>
+        <t t-if="order.order_line.product_id.channel_ids" t-call="website_sale_slides.course_purchased_confirmation_message"/>
     </xpath>
 </template>
 
 <template id="course_purchased_confirmation_message">
     <div>
-        <h4>You have gained access to the following course(s):</h4>
+        <h4 t-if="order.state == 'sale'">You have gained access to the following course(s):</h4>
```

Comment:
As we are in stable, could you maybe keep a single "h4" in case someone has a custom xpath on it ? Otherwise post update this could lead to strange results. Like
```
<h4>
    <t t-if="...">You have gained access ...</t>
    <t t-else="">Once your order ...</t>
</h4>
```